### PR TITLE
Parametrise init container

### PR DESCRIPTION
**Pull request by RafalKorepta**
In some environment publicly available busybox container is not accessible. Now it can be overwritten.

### CODE REVIEW

File: charts/redpanda/values.yaml [hunk 1]
```diff
@@ -511,6 +511,7 @@ statefulset:
   securityContext:
     fsGroup: 101
     runAsUser: 101
+  initContainer: busybox:latest
```
Comment (alejandroEsc) [inline charts/redpanda/values.yaml:514]: Could we make this, for consistency:

```
initContainerImage:
  repository: busybox
  tag: latest
```

**Review by alejandroEsc — COMMENTED** [review]
One suggestion, nothing blocking.

**Comment by alejandroEsc** [comment]
I see this error:

```
create Pod redpanda-jerc6vwp8b-2 in StatefulSet redpanda-jerc6vwp8b failed error: Pod "redpanda-jerc6vwp8b-2" is invalid: spec.initContainers[0].image: Required value                                                                                                                                                                                                9m57s        17      redpanda-jerc6vwp8b.17422b1526c139e8
```

**Comment by RafalKorepta** [comment]
> I see this error:
> 
> ```
> create Pod redpanda-jerc6vwp8b-2 in StatefulSet redpanda-jerc6vwp8b failed error: Pod "redpanda-jerc6vwp8b-2" is invalid: spec.initContainers[0].image: Required value                                                                                                                                                                                                9m57s        17      redpanda-jerc6vwp8b.17422b1526c139e8
> ```

The problem is with the helm command used:
```
helm upgrade redpanda-jerc6vwp8b charts/redpanda --namespace redpanda-jerc6vwp8b --reuse-values --wait --timeout 600s
```

The `--reuse-values` will not work as I introduced new value for init Container:
https://github.com/redpanda-data/helm-charts/pull/326/files#diff-df34075ea48d6120c348021a3727de5f513afd338dfa2e8378993b5b1f374d39R514

File: charts/redpanda/values.yaml (hunk 1 above)
Comment (RafalKorepta) [reply charts/redpanda/values.yaml:514]: Sure

**Comment by alejandroEsc** [comment]
@RafalKorepta could you rebase? I think Joe's PR fixed the reuse value issue.

**Comment by RafalKorepta** [comment]
I already rebased in https://github.com/redpanda-data/helm-charts/compare/eac06b4a53c7aff837f44365536b08ab289f01df..7b5e06ae4e4a56e7d281d305d6e6ad2e7a14c62e push

**Comment by RafalKorepta** [comment]
I made mistake. I changed the values but not the usage of repository and tag variable.

**Review by alejandroEsc — APPROVED** [review]
thank you!

**Pull request merged**